src/Types/BigIntType.php

```diff
@@ -47,15 +47,18 @@ public function convertToPHPValue(mixed $value, AbstractPlatform $platform): int
             return $value;
         }
 
-        if ($value > PHP_INT_MIN && $value < PHP_INT_MAX) {
-            return (int) $value;
-        }
-
         assert(
             is_string($value),
             'DBAL assumes values outside of the integer range to be returned as string by the database driver.',
         );
 
+        if (
+            ($value > PHP_INT_MIN && $value < PHP_INT_MAX)
+            || $value === (string) (int) $value
```

Comment:
I think, we can still get rid of these casts. There are precisely four cases in which our check `$value > PHP_INT_MIN && $value < PHP_INT_MAX` prevents an integer cast. We might as well just enumerate them.

Reply:
Do you mean x86 & x64 min/max?

I wonder why would someone prefer hardcoding values?

TBH, I actually think `$value > PHP_INT_MIN && $value < PHP_INT_MAX` is not ideal as it rounds currently.

Reply:
Very well, lets not over-optimize this.